src/main/groovy/com/palantir/gradle/shadowjar/ShadowJarPlugin.java

```diff
@@ -132,7 +132,12 @@ private void setupShadowJarToShadeTheCorrectDependencies(
         // from another source, so this *should* be ok (there is a test for this).
         ShadowJarVersionLock.excludeConfigurationFromVersionsPropsInjection(project, rejectedFromShading);
 
-        unshaded.getIncoming().beforeResolve(_ignored -> {
+        unshaded.getIncoming().beforeResolve(incoming -> {
+            // do not attempt to process if this is the copy of the configuration - it means that GCV has already
+            // resolved things.
```

Comment:
So here, we're adding a `beforeResolve` on `unshaded`, but then the GCV copies it and that also copies the `beforeResolve` we've set? And then it's that _copied_ `beforeResolve` that tries to illegally modify the original `unshaded` configuration?

Reply:
If this is correct, maybe just expand the comment a bit to cover the copied beforeResolves/what I said above.